File: Java/2D-Arrays/Input2DArray.java
/*
Explanation of Approach:
- The program first takes the dimensions (number of rows and columns) of the 2D array as input from the user.
- It then creates a 2D array with the given dimensions using `int[][] arr = new int[rows][cols];`.
- Next, it enters a nested loop to take input for each element of the 2D array. The user is prompted to enter each element, and the program stores it in the corresponding position of the 2D array.
- Finally, the program outputs the 2D array to the console.

Time Complexity:
- The time complexity of this program is O(rows * cols), where `rows` is the number of rows and `cols` is the number of columns. This is because both the input and output loops run through all the elements of the 2D array once.

Space Complexity:
- The space complexity of this program is O(rows * cols), as it creates a 2D array of dimensions `rows` and `cols`.

Sample Input:
Enter the number of rows: 3
Enter the number of columns: 3
Enter the elements of the 2D array:
1 2 3
4 5 6
7 8 9

Sample Output:
The 2D array is:
1 2 3 
4 5 6 
7 8 9 

In the sample input, the user entered a 3x3 2D array with the elements 1, 2, 3, 4, 5, 6, 7, 8, and 9. The program then displayed the 2D array as output.
*/

import java.util.Scanner;

public class Input2DArray {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        // Input the dimensions of the 2D array
        System.out.print("Enter the number of rows: ");
        int rows = sc.nextInt();
        System.out.print("Enter the number of columns: ");
        int cols = sc.nextInt();

        // Create the 2D array
        int[][] arr = new int[rows][cols];

        // Input the elements of the 2D array
        System.out.println("Enter the elements of the 2D array:");
        for (int i = 0; i < rows; i++) {
            for (int j = 0; j < cols; j++) {
                arr[i][j] = sc.nextInt();
            }
        }

        // Output the 2D array
        System.out.println("The 2D array is:");
        for (int i = 0; i < rows; i++) {
            for (int j = 0; j < cols; j++) {
                System.out.print(arr[i][j] + " ");
            }
            System.out.println();
        }

        sc.close();
    }
}

File: C++/2D-Arrays/Input2DArray.cpp
/*
Explanation of Approach:
- This program takes two integer inputs `rows` and `cols`, which represent the number of rows and columns of the 2D array, respectively.
- It then creates a 2D array `arr` of size `rows` x `cols` to store the elements entered by the user.
- The program uses nested loops to iterate through each element of the 2D array and takes input from the user for each element.
- After taking input, it then outputs the entered 2D array.

Time Complexity:
- The time complexity of this program is O(rows * cols), as we need to input and output each element of the 2D array.

Space Complexity:
- The space complexity of this program is O(rows * cols), as we need to allocate memory for the 2D array of size `rows` x `cols`.

Sample Input:
Enter the number of rows: 3
Enter the number of columns: 4
Enter the elements of the 2D array:
1 2 3 4
5 6 7 8
9 10 11 12

Sample Output:
The entered 2D array is:
1 2 3 4 
5 6 7 8 
9 10 11 12 

In the sample input, the user entered a 3x4 2D array, and the program displayed the entered elements as output.
*/

#include <iostream>

int main() {
    int rows, cols;

    // Input the number of rows and columns
    std::cout << "Enter the number of rows: ";
    std::cin >> rows;
    std::cout << "Enter the number of columns: ";
    std::cin >> cols;

    // Create a 2D array of size rows x cols
    int arr[rows][cols];

    // Input the elements of the 2D array
    std::cout << "Enter the elements of the 2D array:" << std::endl;
    for (int i = 0; i < rows; ++i) {
        for (int j = 0; j < cols; ++j) {
            std::cin >> arr[i][j];
        }
    }

    // Output the 2D array
    std::cout << "The entered 2D array is:" << std::endl;
    for (int i = 0; i < rows; ++i) {
        for (int j = 0; j < cols; ++j) {
            std::cout << arr[i][j] << " ";
        }
        std::cout << std::endl;
    }

    return 0;
}

